### PR TITLE
Enable dynamic updates

### LINKED ISSUE #27: Value not updating dynamically
I am using ember-prism in my ember application.
`{{#code-block language="json"}}
{{jsonData}}
{{/code-block}}`
When value of `jsonData` changes it does not render in {{code-block}}, but I can see it changed when I print jsonData on the page without {{code-block}} by just using {{jsonData}}.

### PR DESCRIPTION
**Pull request by simonihmig**
Introduces a inline invocation `{{code-block code=code}}` that supports dynamic updates.
Found a way to also support it for the block form, but requires some tricks, so inline form seems preferable.
Added integration tests.

Closes #27

### CODE REVIEW

**Comment by simonihmig** [comment]
@rwwagner90 Still need to update the docs a bit, will do that if the changes here are approved!

File: addon/components/code-base.js [hunk 1]
```diff
@@ -1,21 +1,65 @@
...
+  },
+
+  didRender() {
+    this._super(...arguments);
+    let code = this.get('code');
+    if (this.get('hasBlock')) {
+      code = this.getBlockContent();
+
+      // if block content has changed, force reevaluation of `prismCode`
+      if (code !== this._lastBlockContent) {
```
Comment (rwwagner90) [inline addon/components/code-base.js]: Should  wee maybe use `didUpdateAttrs` here?

File: index.js [hunk 2]
```diff
@@ -57,6 +57,8 @@ module.exports = {
       }
     }
 
+    app.import('vendor/ember-prism.js');
```
Comment (rwwagner90) [inline index.js]: What is this for?

File: addon/components/code-base.js (hunk 1 above)
Comment (simonihmig) [reply addon/components/code-base.js]: This is used to detect changes to the component's block content, and not any props passed to the component, so I don't think `didUpdateAttrs` will be called in this case, does it?

File: index.js (hunk 2 above)
Comment (simonihmig) [reply index.js]: We need to make prism.js go into "manual" mode, i.e. not apply highlighting to any code elements by default on `DOMContentLoaded`, see https://github.com/PrismJS/prism/blob/6068bf18e53222a1ee712b073a9864413fefeef3/prism.js#L531-L541. Mangling with the DOM on code blocks rendered with Glimmer will cause exceptions when later Glimmer tries to update "its" DOM. Which is the case for the dynamic updates enabled here. See also the comment in the imported file.

As we let Glimmer render the *already highlighted* code (here: https://github.com/shipshapecode/ember-prism/pull/33/files#diff-28fe0c00eea01c10c0b6b9d67c35f741R31), there is no need for this highlight by default behaviour!

**Comment by simonihmig** [comment]
Tests are failing due to `ember-in-element-polyfill` requiring node 8+, so dropped node 6 support here: https://github.com/shipshapecode/ember-prism/pull/34

File: addon/components/code-base.js (hunk 1 above)
Comment (rwwagner90) [reply addon/components/code-base.js]: I'm not sure. I would have to try it out. Just trying to make things as simple as we can

Comment (simonihmig) [reply addon/components/code-base.js]: I quickly tried with `didReceiveAttrs`, but that definitely does not work. It is called before rendering, so not able to query the DOM correctly. `didRender` should be fine IMO!

**Comment by simonihmig** [comment]
Rebased, Travis is happy now!

**Pull request merged**